tilavarauspalvelu/migrations/0015_determine_preferred_language.py

```diff
@@ -0,0 +1,33 @@
+from django.db import migrations, models
+from django.db.models.functions import Coalesce
+
+
+def determine_preferred_language(apps, schema_editor):
+    User = apps.get_model("tilavarauspalvelu", "User")
+    Reservation = apps.get_model("tilavarauspalvelu", "Reservation")
+
+    User.objects.filter(preferred_language=None).alias(
+        latest_reservation_language=Coalesce(
+            # Use the latest reservation language
+            models.Subquery(
+                Reservation.objects.filter(user=models.OuterRef("pk"))
```

Comment:
It now occurs to me that the reservations should be further narrowed to `type = NORMAL` (made from the client UI where lang could be selected) and `reservee_language` not empty (just in case, although it probably doesn't matter with the previous condition).

Reply:
Added these two filters